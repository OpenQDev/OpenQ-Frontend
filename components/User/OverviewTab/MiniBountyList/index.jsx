// Third Party
import React from 'react';

// Custom
import MiniBountyCard from '../MiniBountyCard';

const MiniBountyList = ({ title, payouts }) => {
  return (
    <div className='px-8 py-6 pb border-t border-web-gray'>
      <h2 className='font-semibold text-lg'>Claimed Bounties {title}</h2>
      <div>
        {payouts.length != 0 ? (
          <ul>
            {payouts.map((payout, index) => {
              return <MiniBountyCard key={index} payout={payout} />;
            })}
          </ul>
        ) : (
          <span className='pt-2'>No Bounties {title}</span>
        )}
      </div>
    </div>
  );
};
export default MiniBountyList;
