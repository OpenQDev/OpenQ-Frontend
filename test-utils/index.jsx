// Third party
import React from 'react';
// Cusotm
import { render, renderHook } from '@testing-library/react';
import { cleanup } from '@testing-library/react';

import StoreProvider from '../store/Store/TestStoreProvider';
import AuthProvider from '../store/AuthStore/TestAuthProvider';
import FundProvider from '../components/FundBounty/FundStore/TestFundProvider';
import TokenProvider from '../components/TokenSelection/TokenStore/TokenProvider';

// Add in any providers here if necessary:
// test-utils.js
// (ReduxProvider, ThemeProvider, etc)
afterEach(() => {
  cleanup();
});

const customRender = (ui, options = {}, storeProps, authProps) => {
  const IntersectionObserverMock = vi.fn(() => ({
    disconnect: vi.fn(),
    observe: vi.fn(),
    takeRecords: vi.fn(),
    unobserve: vi.fn(),
  }));
  vi.stubGlobal('IntersectionObserver', IntersectionObserverMock);
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <TokenProvider>{children}</TokenProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return render(ui, { wrapper: Providers, ...options });
};
const customHookRender = (hook, options = {}, storeProps, authProps, fundProps) => {
  const Providers = ({ children }) => {
    return (
      <AuthProvider StoreProps={authProps}>
        <StoreProvider StoreProps={storeProps}>
          <FundProvider StoreProps={fundProps}>
            <TokenProvider>{children}</TokenProvider>
          </FundProvider>
        </StoreProvider>
      </AuthProvider>
    );
  };

  return renderHook(hook, { wrapper: Providers, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render, customHookRender as hookRender };
