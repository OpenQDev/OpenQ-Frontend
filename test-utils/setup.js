import { expect, beforeEach } from 'vitest';
import { cleanup } from '@testing-library/react';
import matchers from '@testing-library/jest-dom/matchers';

expect.extend(matchers);
beforeEach(() => {
  cleanup();
});
process.on('unhandledRejection', (reason) => {
  // eslint-disable-next-line no-console
  console.log(`FAILED TO HANDLE PROMISE REJECTION`);
  throw reason;
});
