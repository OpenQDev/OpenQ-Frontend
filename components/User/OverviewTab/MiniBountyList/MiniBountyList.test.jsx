/**
 * @vi-environment jsdom
 */
import React from 'react';

import { render, screen } from '../../../../test-utils';
import MiniBountyList from '../../../../components/User/OverviewTab/MiniBountyList';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('MiniBountyList', () => {
  const payouts = [
    {
      bounty: {
        bountyId: 'I_kwDOGWnnz85I9Agl',
        issuer: {
          id: '0x46e09468616365256f11f4544e65ce0c70ee624b',
        },
      },
      volume: '600000000000000000',
      tokenAddress: '0x0000000000000000000000000000000000000000',
    },
    {
      bounty: {
        bountyId: 'I_kwDOGWnnz85I9Agl',
        issuer: {
          id: '0x46e09468616365256f11f4544e65ce0c70ee624b',
        },
      },
      volume: '900000000000000000',
      tokenAddress: '0x0000000000000000000000000000000000000000',
    },
  ];

  it('should match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<MiniBountyList payouts={payouts} title={'example'} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  it('should render MiniBountyList with link', async () => {
    // ARRANGE
    render(<MiniBountyList payouts={payouts} title={'example'} />);

    // ASSERT
    expect(await screen.findByText(/0.40/)).toBeInTheDocument();
    expect(await screen.findByText(/0.60/)).toBeInTheDocument();

    // should not have null or undefined values
    const nullish = [...screen.queryAllByRole(/null/), ...screen.queryAllByRole(/undefined/)];
    expect(nullish).toHaveLength(0);
  });
});
