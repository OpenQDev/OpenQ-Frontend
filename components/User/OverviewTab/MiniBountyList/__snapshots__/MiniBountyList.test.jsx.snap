// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MiniBountyList > should match DOM Snapshot 1`] = `
<div
  className="px-8 py-6 pb border-t border-web-gray"
>
  <h2
    className="font-semibold text-lg"
  >
    Claimed Bounties 
    example
  </h2>
  <div>
    <ul>
      <MiniBountyCard
        payout={
          {
            "bounty": {
              "bountyId": "I_kwDOGWnnz85I9Agl",
              "issuer": {
                "id": "0x46e09468616365256f11f4544e65ce0c70ee624b",
              },
            },
            "tokenAddress": "0x0000000000000000000000000000000000000000",
            "volume": "600000000000000000",
          }
        }
      />
      <MiniBountyCard
        payout={
          {
            "bounty": {
              "bountyId": "I_kwDOGWnnz85I9Agl",
              "issuer": {
                "id": "0x46e09468616365256f11f4544e65ce0c70ee624b",
              },
            },
            "tokenAddress": "0x0000000000000000000000000000000000000000",
            "volume": "900000000000000000",
          }
        }
      />
    </ul>
  </div>
</div>
`;
